### PR TITLE
Support dropping non-Raft nodes

### PR DESCRIPTION
**Pull request by otoolep**
With this change dropping non-Raft nodes is supported.

### CODE REVIEW

**Comment by otoolep** [comment]
This is still work-in-progress, and is mostly code from @corylanou 

The idea here is that we get incremental support out for this command, for very specific situations.

cc @jwilder

File: meta/data.go [hunk 1]
```diff
@@ -74,14 +74,79 @@ func (data *Data) CreateNode(host string) error {
...
+	if id == 0 {
+		return ErrNodeIDRequired
+	}
+	// Is this a valid node?
+	nodeInfo := data.Node(id)
+	if nodeInfo == nil {
+		return ErrNodeNotFound
+	}
+	// Only dropping non-Raft nodes currently supported.
+	peers, err := e.Store.Peers()
```
Comment (otoolep) [inline meta/data.go]: This doesn't work, but it's the check I want to add. Once this check is in place, we can forget about all the Raft complexity until later.

**Comment by otoolep** [comment]
Started with https://github.com/influxdb/influxdb/pull/4233.

This an minimum viable fix for the "extra" nodes reporting in https://github.com/influxdb/influxdb/issues/4212.

**Comment by otoolep** [comment]
I'd like feedback from @corylanou and @jwilder

**Comment by otoolep** [comment]
Raft-check updated to take place at a higher-level, code now compiles.

**Comment by otoolep** [comment]
Tested:
```
> show servers
id      cluster_addr    raft    raft-leader
1       localhost:8088  true    true
2       localhost:8188  true    false
3       localhost:8288  true    false

> drop server 3
ERR: node is a Raft node
Warning: It is possible this error is due to not setting a database.
Please set a database with the command "use <database>".
> 
```

**Comment by otoolep** [comment]
Patch ready for merging, 4 green CI builds.

https://circleci.com/gh/influxdb/influxdb/tree/drop_node_non_raft

File: meta/statement_executor_test.go [hunk 2]
```diff
@@ -141,6 +144,35 @@ func TestStatementExecutor_ExecuteStatement_ShowServers(t *testing.T) {
...
+		return &meta.NodeInfo{
+			ID: 1, Host: "node1",
+		}, nil
+	}
+
+	// Ensure Raft nodes cannot be dropped.
+	e.Store.PeersFn = func() ([]string, error) {
+		return []string{"node1"}, nil
+	}
+	if res := e.ExecuteStatement(influxql.MustParseStatement(`DROP SERVER 1`)); res.Err == nil || res.Err.Error() != "node is a Raft node" {
```
Comment (corylanou) [inline meta/statement_executor_test.go]: Could use `ErrNodeRaft` here to lessen the brittleness of this test.

**Comment by corylanou** [comment]
This looks good.  Limiting it to non-raft nodes takes out most of the real hard problems and allows us to move forward in another PR to do it right.

+1

**Comment by toddboom** [comment]
:+1: for getting this out as an intermediate step.

**Comment by otoolep** [comment]
Made the change suggested by @corylanou -- merging on green.

**Pull request merged**